### PR TITLE
Add rollback tests to CI pipeline

### PR DESCRIPTION
**Pull request by LTA-Thinking**
## Description
Adds rollback tests to CI pipeline

## Related issues
Addresses [User Story 113052](https://microsofthealth.visualstudio.com/Health/_workitems/edit/113052): Add rollback tests to OSS CI

## Testing
Ran the CI pipeline with the new definition

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [x] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Skip

### CODE REVIEW

File: build/ci-pipeline.yml [hunk 1]
```diff
@@ -180,58 +340,56 @@ stages:
...
   jobs:
   - template: ./jobs/run-tests.yml
     parameters:
       version: R5
-      keyVaultName: $(DeploymentEnvironmentNameR5)
-      appServiceName: $(DeploymentEnvironmentNameR5)
+      keyVaultName: $(DeploymentEnvironmentName)
+      appServiceName: $(DeploymentEnvironmentName)
 
+# *********************** Finalize ***********************
```
Comment (mikaelweave) [inline build/ci-pipeline.yml:392]: Should the docker main tag be added after the rollback tests? Or is the goal to provide artifacts before the rollback tests run?

**Review by mikaelweave — COMMENTED** [review]
One question about the pipeline before further review. Excited about this so I know I am NOT making breaking schema changes hahahaa. Great job on this. 😄

File: build/ci-pipeline.yml (hunk 1 above)
Comment (mikaelweave) [reply build/ci-pipeline.yml:392]: That makes sense.

Leaving it as is would make a different type of sense where we'd get faster availability of the artifact if needed downstream, but we would need a manual check to ensure the pipeline passed.

File: build/ci-pipeline.yml [hunk 2]
```diff
@@ -20,6 +21,39 @@ stages:
...
+    - task: AzurePowerShell@5
+      displayName: 'Azure PowerShell script: InlineScript'
+      inputs:
+        azureSubscription: $(ConnectedServiceName)
+        azurePowerShellVersion: latestVersion
+        ScriptType: inlineScript
+        Inline: |
+          $testNamePatterns = @("SNAPSHOT*","FHIRCOMPATIBILITYTEST*","FHIRINTEGRATIONTEST*","FHIRRESOURCECHANGEDISABLEDTEST*","BASE*","SNAPSHOT*")
+          foreach ($pattern in $testNamePatterns) {
+            $resources = Get-AzResource -ResourceGroupName $(ResourceGroupName) -ResourceType 'Microsoft.Sql/servers/databases' -Name $pattern
```
Comment (mikaelweave) [inline build/ci-pipeline.yml:50]: Question - is this a uniquely SQL problem or do we need something for Cosmos too?

**Review by mikaelweave — APPROVED** [review]
One open question, no changes needed. Great work on this. It will increase my confidence in releasing any SQL schema changes. 🦾

We should also monitor if the longer pipeline run time causes any issue. We *could* parallelize but that would make the pipeline too complex to read on one screen haha. I have some ideas to improve integration/E2E test reliability - if we get this in I don't think it'll be an issue.

File: build/ci-pipeline.yml (hunk 2 above)
Comment (LTA-Thinking) [reply build/ci-pipeline.yml:50]: It is just a SQL issue. I don't know exactly why, but the equivalent process in Cosmos doesn't leave a bunch of left over containers.

**Pull request merged**